empress/recon_vis/recon_viewer.py

```diff
@@ -1,80 +1,206 @@
+    :recon_dict: Reconciliation represented in dictionary format
     """
-    # convert host and parasite dicts to objects and populate nodes with their temporal order
-    host_tree, parasite_tree, consistency_type = utils.build_trees_with_temporal_order(host_dict, parasite_dict, recon_dict)
-    recon = utils.dict_to_reconciliation(recon_dict, event_frequencies)
+    host_tree, parasite_tree, recon, consistency_type = utils.convert_to_objects(host_dict, parasite_dict, recon_dict, event_frequencies)
+    #Checks to see if the trees(or reconciliation) are empty
+    if host_tree is None or parasite_tree is None or recon is None:
+        return None
+
+    fig = plot_tools.FigureWrapper(TREE_TITLE, axes)
+    create_legend(fig, consistency_type)    
+
+    #Calculates font sizes
+    num_tips = len(host_tree.leaf_list) + len(parasite_tree.leaf_list)
+    num_nodes = len(host_tree.postorder_list) + len(parasite_tree.postorder_list)
+    tip_font_size, internal_font_size = calculate_font_size(num_tips, num_nodes)
 
-    fig = plot_tools.FigureWrapper(consistency_type, axes)
-    if consistency_type != utils.ConsistencyType.NO_CONSISTENCY:
-        render_host(fig, host_tree, show_internal_labels)
-        host_lookup = host_tree.name_to_node_dict()
-        render_parasite(fig, parasite_tree, recon, host_lookup, show_internal_labels, show_freq)
-    return fig
+    root = parasite_tree.root_node
+    host_lookup = host_tree.name_to_node_dict()
+    parasite_lookup = parasite_tree.name_to_node_dict()
+    
+    #Populate Host Nodes with track count
+    populate_host_tracks(root, recon, host_lookup)
+    
+    #Render Host Tree
+    render_host(fig, host_tree, show_internal_labels, tip_font_size, internal_font_size)
+    
+    #Sets the offsets between tracks on each host node
+    set_offsets(host_tree)
+    
+    #Render Parasite Tree
+    render_parasite(fig, parasite_tree, recon, host_lookup, parasite_lookup, show_internal_labels, show_freq, tip_font_size, internal_font_size)
+
+    return fig 
 
+def create_legend(fig, consistency_type):
+    legend_elements = [
+                        Line2D([0], [0], marker= COSPECIATION_NODE_SHAPE, color='w', label='Cospeciation', \
+                               markerfacecolor=COSPECIATION_NODE_COLOR, markersize=NODESIZE),
+                        Line2D([0], [0], marker=DUPLICATION_NODE_SHAPE, color='w', label='Duplication', \
+                               markerfacecolor=DUPLICATION_NODE_COLOR, markersize=NODESIZE),
+                        Line2D([0], [0], marker=TRANSFER_NODE_SHAPE, color='w', label='Transfer', \
+                               markerfacecolor=TRANSFER_NODE_COLOR, markersize=NODESIZE),\
+                        LineCollection( [[(0, 0)]], linestyles = ['dashed'], \
+                                       colors = [LOSS_EDGE_COLOR], label='Loss')
+                      ]
+    fig.set_legend(legend_elements, title = consistency_type)
+def set_offsets(tree):
```

Comment:
Add space between functions

Reply:
Commenting "Done" so I can keep track of what has been fixed already.